crates/runtime/src/component/dataset.rs

```diff
@@ -344,6 +353,56 @@ impl Dataset {
         }
     }
 
+    /// For [`Dataset`]s where the path in the `from` field is a [`TableReference`], parse and return the [`TableReference`].
+    ///
+    ///
+    #[must_use]
+    pub fn table_reference_path(
```

Comment:
Implementation LGTM and align to DF. Not sure this should be part of Dataset component - might be better to add as utility function so it can be re-used
https://github.com/apache/datafusion/blob/main/datafusion/common/src/utils/mod.rs#L276